lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorManager.java

```diff
@@ -306,6 +306,7 @@ public void onEvaluatorException(final EvaluatorException exception) {
           this.messageDispatcher.onDriverRestartEvaluatorFailed(failedEvaluator);
```

Comment:
@johnyangk I believe the logic is parallel, so it should be fine to call ``messageDispatcher.close()``. Thanks!